streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -550,6 +590,8 @@ private static void verifyAllTransactionFinished(final KafkaConsumer<byte[], byt
         producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
         producerProps.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, true);
 
+        final Map<TopicPartition, Long> endMarkerOffset = new HashMap<>();
```

Comment:
As there might be pending transaction, we need to improve the way how we verify that all transaction are finished. For this, we need to remember the offsets of our "topic end marker messages".

Reply:
See my question above: could we just rely on the maintained `offsets` map last values?